index.rst

```diff
@@ -5,77 +5,331 @@
+This is, for instance, the current approach implemented by the ``Watcher``.
+Any new alarm requires code changes. 
+
+Analysis Process
+----------------
+
+A key functionality of the ``Catcher``, that differentiates it from the ``Watcher``, is the ability to execute an analysis process once a certain condition is met.
+In general, we assume that these analysis will consist of lightweight computations.
+Whenever possible, the computation should be offloaded to the OCPS.
+It is likely that the data processed by the rule will be relevant to the downstream processing, so it should  be made available to it as well.
+
+These processing routines should be developed in the ``Catcher`` codebase using a common framework, similar to that provided by SAL Scripts (e.g. a base class with a well defined interface).
+It is also important that the results/outputs of a process follow a well defined format that can be consumed downstream.
+
+.. _introduction-process-image-data:
 
 Process Image Data
 ------------------
 
-Another important feature of the Catcher was the capability to process camera image data using the OCPS.
-The OCPS is a CSC that is capable of executing DM-pipeline processes.
+Another important feature of the ``Catcher`` is the capability to process camera image data using the ``OCPS``.
+The ``OCPS`` is a CSC that is capable of executing DM-pipeline processes.
 It is going to be key in allowing remote processing of Main Camera data during commissioning and operations.
 
-Nevertheless, if the Catcher is not a CSC it won't be able to interact direcly with OCPS to process image data.
-One alternative would be to interact with the OCPS remote backend direcly, instead of going through the CSC.
-For that, the OCPS backend implements a REST API that we could leverage to post requests and receive responses back.
+Since the ``Catcher`` is a CSC it can interact directly with ``OCPS`` to process image data, by sending commands to it.
+
+Alarms
+------
+
+Depending of the results of the analysis process the ``Catcher`` should be able to generate alerts to the system.
+For this we can follow a similar approach to that already implemented in the ``Watcher``.
+In short, the ``Catcher`` will have an "alarm" event with payloads to indicate the "severity" and other relevant information.
+There will also be commands to mute and acknowledge these alarms.
+We can also consider to borrow the alarm escalation model from the ``Watcher``.
 
 Generate reports
 ----------------
 
-The Catcher should also be able to generate reports, with graphs and performing additional lightweight computation.
-The original idea was to rely on parameterized Jupyter notebooks that would later be converted to webpages and rendered online so users could inspect the results, and use as reference for additional followup.
+The ``Catcher`` should also be able to generate reports, with graphs and additional lightweight computation.
```

Comment:
you want the computations in the report?
Also, add an `s` on computation..

Reply:
I think there might be cases where we might need to do some additional computations.. like fits/extrapolations/etc..